nilearn/decomposition/multi_pca.py

```diff
@@ -360,3 +372,44 @@ def inverse_transform(self, component_signals):
         # XXX: dealing properly with 2D/ list of 2D data?
         return [nifti_maps_masker.inverse_transform(signal)
                 for signal in component_signals]
+
+    def group_pca(self, subject_pcas):
+        """Compute the group PCA from individual subject PCAs"""
+        if self.incremental_group_pca:
+            return self._incremental_group_pca(subject_pcas)
+        else:
+            return self._in_memory_group_pca(subject_pcas)
+
+    def _incremental_group_pca(self, subject_pcas):
+        def _incremental_group_pca_impl(subject_pcas, n_components):
+            # delayed import because IncrementalPCA is only supported for
+            # sklearn versions > 0.16
+            from sklearn.decomposition.incremental_pca import IncrementalPCA
+
+            pca = IncrementalPCA(n_components=self.n_components)
+            for subject_pca in subject_pcas:
+                pca.partial_fit(subject_pca)
+
+            return pca.components_, pca.singular_values_
+
+        return cache(_incremental_group_pca_impl,
+                     self.memory,
+                     func_memory_level=3,
+                     memory_level=self.memory_level)(
+                         subject_pcas, self.n_components)
+
```

Comment:
Is this the canonical way of hiding cached functions ? This is not what is used in other parts of nilearn. It looks clean though